src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -495,15 +496,21 @@ async function validate_bucket_args(config_fs, data, action) {
  * When setting notifications, we are supposed to send a test notification.
  * If this test fails, we fail the user's request.
  * @param {object} config_fs
- * @param {object} notifications bucket's notification conf
+ * @param {object} user_input user's input, including new notifications, if any
  * @returns {Promise} Error encountered during notification test, if any
  */
-async function validate_bucket_notifications(config_fs, notifications) {
-    if (!notifications) {
+async function validate_bucket_notifications(config_fs, user_input) {
+    if (!user_input.notifications) {
         return;
     }
     //test_notification returns an error if notification fails for the given config
-    const test_notif_err = await notifications_util.test_notifications(notifications, config_fs.config_root);
+    const test_notif_err = await notifications_util.test_notifications(
+        user_input.notifications,
+        config_fs.config_root,
+        {
+            params: {bucket: user_input.name},
+            request_id: crypto.randomUUID().toString()
```

Comment:
why we need request_id ?

Reply:
It's the structure aws' test notification uses (https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html).
Note that this is only for NC, for containerized we do have a "real" request id.